pager/src/main/java/dev/chrisbanes/accompanist/pager/PagerState.kt

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:Suppress("MemberVisibilityCanBePrivate")
+
+package dev.chrisbanes.accompanist.pager
+
+import android.util.Log
+import androidx.annotation.FloatRange
+import androidx.annotation.IntRange
+import androidx.compose.animation.core.AnimationSpec
+import androidx.compose.animation.core.AnimationState
+import androidx.compose.animation.core.DecayAnimationSpec
+import androidx.compose.animation.core.animate
+import androidx.compose.animation.core.animateDecay
+import androidx.compose.animation.core.calculateTargetValue
+import androidx.compose.animation.core.spring
+import androidx.compose.foundation.gestures.DraggableState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.Saver
+import androidx.compose.runtime.saveable.listSaver
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import kotlin.math.absoluteValue
+import kotlin.math.floor
+import kotlin.math.roundToInt
+
+private const val LogTag = "PagerState"
+
+/**
+ * Creates a [PagerState] that is remembered across compositions.
+ *
+ * Changes to the provided initial values will **not** result in the state being recreated or
+ * changed in any way if it has already been created.
+ *
+ * @param currentPage the initial value for [PagerState.currentPage]
+ * @param currentPageOffset the initial value for [PagerState.currentPageOffset]
+ * @param pageCount the initial value for [PagerState.pageCount]
+ */
+@ExperimentalPagerApi
+@Composable
+fun rememberPagerState(
+    currentPage: Int = 0,
+    @FloatRange(from = 0.0, to = 1.0) currentPageOffset: Float = 0f,
+    pageCount: Int = 0,
```

Comment:
I think this param should be the first one and have no default value. this component makes no sense with no pages

Reply:
Good catch, done.